### PR TITLE
[14.0][MIG] l10n_th_base_location

### PR DESCRIPTION
**Pull request by Saran440**
- Change onchange to compute function
- Test Script (Currenly, Belgium can't download address file zip from website)

### CODE REVIEW

File: l10n_th_base_location/models/res_partner.py [hunk 1]
```diff
@@ -7,10 +7,10 @@
...
     _inherit = "res.partner"
 
-    @api.onchange("zip_id")
-    def _onchange_zip_id(self):
-        res = super()._onchange_zip_id()
-        if self.zip_id and self.country_id.code == "TH":
-            address = self.zip_id.city_id.name.split(", ")
-            self.update({"street2": address[0], "city": address[1]})
-        return res
+    @api.depends("zip_id")
```
Comment (kittiu) [inline l10n_th_base_location/models/res_partner.py:10]: You changed this to compute. You will need to add param `compute="_compute_zip_id` and `readonly=False` to the related fields also, right?

Please correct me if I am wrong about the new compute field.

**Comment by kittiu** [comment]
@Saran440 can you make sure the improvement https://github.com/OCA/l10n-thailand/pull/225 is applied here too.

**Comment by kittiu** [comment]
@Saran440 ca

> @Saran440 can you make sure the improvement #225 is applied here too.

ping @Saran440

File: l10n_th_base_location/models/res_partner.py (hunk 1 above)
Comment (Saran440) [reply l10n_th_base_location/models/res_partner.py:10]: In module `base_location_geonames_import` has change structure city fields to compute field.
then, We will change it to compute too.

**Review by kittiu — APPROVED** [review]

**Pull request merged**